ceph/ceph_admin/__init__.py

```diff
@@ -170,13 +170,32 @@ def check_exist(self, daemon, ids, timeout=None, interval=5):
 
         return False
 
-    def shell(self, remote, args, **kwargs):
+    def get_role_service(self, service_name):
+        """
+        Get service info by name
+
+        Args:
+            service_name: service name
+        Returns:
+            service
+        """
+        out, _ = self.shell(
+            remote=self.installer,
+            args=["ceph", "orch", "ls", "-f", "json-pretty"],
+        )
+        for svc in json.loads(out):
+            if service_name in svc.get("service_name"):
+                return svc
+        return None
```

Comment:
I like the idea of returning `None`. Could you evaluate if we should be raising an exception here? Like ResourceNotFound

Reply:
just not raising exception here, currently handled in the called function to address negative cases.
https://github.com/red-hat-storage/cephci/blob/d47106796aaec474e28b6a85e6f33156aec83b27/ceph/ceph_admin/apply.py#L60